File: Frontend/src/app/Interfaces/topic.ts
export interface TopicDetaild {
    topicId:number,
    title: string,
    description: string,
    topics: number,
    parentId: number|null,
    courseId?:number
}
export interface subTopic {
  subTopicId:number,
  title: string,
  description: string,
  topics: number,
  parentId: number|null
}

export interface TopicInput{
  title: string,
  description: string,
  topics: number,
  parentId: number|null
  courseId?: number
}

export interface Topic{
  topicId: number,
    title: string,
    description: string,
    subtopics: Array<SubTopic>
  }
  export interface SubTopic {
    subTopicId:number,
    title: string,
    description: string,
    media: []
  }
  export interface Media{
    mediaId: number,
    topicId: number,
    mediaSrc: string,
    mediaType:string
  }

File: Udemy/Udemy/Models/Topic.cs
using System;
using System.Collections.Generic;

namespace Udemy.Models;

public partial class Topic
{
    public int TopicId { get; set; }

    public string Title { get; set; } = null!;

    public string Description { get; set; } = null!;

    public int? CourseId { get; set; }

    public int? CreatedBy { get; set; }

    public int? ModifiedBy { get; set; }

    public DateTime? CreatedDate { get; set; }

    public DateTime? ModifiedDate { get; set; }

    public virtual Course? Course { get; set; }

    public virtual User? CreatedByNavigation { get; set; }

    public virtual User? ModifiedByNavigation { get; set; }

    public virtual ICollection<Progress> Progresses { get; set; } = new List<Progress>();

    public virtual ICollection<SubTopic> SubTopics { get; set; } = new List<SubTopic>();
}
